core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -86,6 +86,20 @@ public boolean isEmpty() {
     return (globalDeletes == null || globalDeletes.length == 0) && sortedDeletesByPartition.isEmpty();
   }
 
+  public List<DeleteFile> referencedDeleteFiles() {
+    List<DeleteFile> deleteFiles = Lists.newArrayList();
```

Comment:
Optional comment: small optimization can be done by knowing the initial length, and checking isEmpty
```
if (isEmpty()) {
   return Lists.empty()
else
   List<DeleteFile> deleteFiles = Lists.newArrayList(globalDeletes.length + sortedDeletesByPartition.length);
   ...
```

Reply:
Will do!

Reply:
I was about to implement this but then I realized `sortedDeletesByPartition.length` is not accurate as each entry contains an array of delete files. In order to compute a right estimate, I'd need to iterate through the map.